src/main/java/org/springframework/data/jpa/repository/support/QueryHints.java

```diff
@@ -17,21 +17,20 @@
 
 import java.util.Collections;
 import java.util.Iterator;
-import java.util.Map;
-import java.util.Map.Entry;
+import java.util.List;
 
 import javax.persistence.EntityManager;
 
 /**
- * QueryHints provides access to query hints defined via {@link CrudMethodMetadata#getQueryHints()} by default excluding
- * JPA {@link javax.persistence.EntityGraph}.
+ * QueryHints provides access to query hints defined via {@link CrudMethodMetadata#getQueryHintList()} by default
+ * excluding JPA {@link javax.persistence.EntityGraph}.
  *
  * @author Christoph Strobl
  * @author Oliver Gierke
  * @author Jens Schauder
  * @since 2.0
  */
-interface QueryHints extends Iterable<Entry<String, Object>> {
+interface QueryHints extends Iterable<QueryHintValue> {
```

Comment:
We could either remove `Iterable` entirely or use `Entry<String, List<Object>>`. Since the type is internal anyway, we're free for a bit more aggressive cleanup.